src/mappings/Reputation.ts

```diff
@@ -0,0 +1,54 @@
+import 'allocator/arena'
+export { allocate_memory }
+
+import { Entity, Address, Value, store, crypto, ByteArray, U256 } from '@graphprotocol/graph-ts'
+
+import { Reputation, Mint, Burn } from '../types/Reputation/Reputation'
+import { concat, zero256, isZero } from '../utils'
+
+function update(contract: Address, owner: Address): void {
+    let rep = Reputation.bind(contract);
+    let ent = new Entity();
+    let id = crypto.keccak256(concat(contract, owner)).toHex();
+    ent.setString('id', id);
+    ent.setAddress('contract', contract);
+    ent.setAddress('address', owner);
+    let balance = rep.balanceOf(owner);
```

Comment:
Here it validate the state (balance) via the blockchain (which can be calculate off chain by adding /reducing balances.
Pros : 
1. Additional validation against the blockchain.
Cons: 
1. Redundant web3 call . (time) 

questions : 
- Should we consider adding the balance to the Mint/Burn events ?
- How the performance will be effected by reading the state from the bc ? let say you have multiple Mint/Burn events on the same block.

Reply:
The main concern in my opinion is that we don't want to replicate the entire logic of the contract inside the cache. This would make the cache pretty brittle.
answers:
- I don't think we need to add the balance to the events.
- I don't know how `graph-node` actually implements those web3 calls, they may do it in parallel across mappings which means its not probably going to be a problem.